apis/configwatch.go

```diff
@@ -181,3 +209,209 @@ func MonitorTLS(start time.Time, expiry time.Time, conf *config.Config) {
 		}
 	}
 }
+
+func (s *Server) getClientValidator(helloInfo *tls.ClientHelloInfo) func([][]byte, [][]*x509.Certificate) error {
+
+	log.Debug().Msgf("Inside get client validator - %v", helloInfo.Conn.RemoteAddr())
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		//added DNSName
+		opts := x509.VerifyOptions{
+			Roots:         s.CaCertPool,
+			CurrentTime:   time.Now(),
+			Intermediates: x509.NewCertPool(),
+			KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		}
+		_, err := verifiedChains[0][0].Verify(opts)
+		if err != nil {
+			log.Error().Err(err).Msgf("certs verification failed")
+			return err
+		}
+
+		if len(s.DNSName) == 0 {
+			log.Info().Msgf("dnsname is undefined")
+			return nil
+		}
+		candidateName := toLowerCaseASCII(s.DNSName) // Save allocations inside the loop.
+		validCandidateName := validHostnameInput(candidateName)
```

Comment:
I think it makes sense to drop this and just have a check on validHostnameInput on the SANs from the cert.  Here we are validating essentially the s.DNSName configuration value only.  I think it is more important to actually check the SANs from the client cert and ensure it is RFC compliant

Reply:
check here should be for valid regex or hostname provided in the config file.